lib/stellar_core_commander/docker_process.rb

```diff
@@ -206,18 +206,44 @@ def shared_history_volume
     end
 
     Contract None => String
-    def history_commands
-      if use_s3
-        <<-EOS.strip_heredoc
-          HISTORY_GET=aws s3 --region #{@s3_history_region} cp #{@s3_history_prefix}/%s/{0} {1}
-          HISTORY_PUT=aws s3 --region #{@s3_history_region} cp {0} #{@s3_history_prefix}/%s/{1}
-        EOS
+    def history_get_command
+      cmds = {}
+      @quorum.each do |q|
+        if q == @name
+          next
+        end
+        cmd = ""
+        if SPECIAL_PEERS.has_key? q
+          cmd = SPECIAL_PEERS[q][:get]
+        elsif use_s3
+          cmd = "aws s3 --region #{@s3_history_region} cp #{@s3_history_prefix}/%s/{0} {1}"
+        else
+          cmd = "cp /history/%s/{0} {1}"
+        end
+        cmds[cmd] = ()
```

Comment:
Could do. I'm comfortable using dicts as sets in local cases like this but if you prefer I'm happy to import it and use Set.